src/registry2.0/tests/entry.tcl

```diff
@@ -74,6 +74,9 @@ proc main {pextlibname} {
         test_set {[registry::entry search name -glob vi*]} {$vim1 $vim2 $vim3}
         test_set {[registry::entry search name -regexp {zlib|pcre}]} \
             {$zlib $pcre}
+
+        # test that passing in confusing arguments doesn't crash
+        test {[catch {[registry::entry search name vim1 --]}] == 1}
```

Comment:
The innermost pair of square brackets are a syntax error. This means catch will return 1 even if the args are something valid.

Reply:
Thanks, I think I've fixed this now.